.github/workflows/cloud-regress.yml

```diff
@@ -73,15 +83,68 @@ jobs:
           path: /tmp/neon/
           prefix: latest
 
+      - name: Get endpoint ID
+        id: endpoint-id
+        uses: ./.github/actions/neon-get-endpoint
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+
+      - name: Get the connection URI
+        id: connect-uri
+        uses: ./.github/actions/neon-get-connection-uri
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+          endpoint_id: ${{steps.endpoint-id.outputs.id}}
+
+      - name: Create a new branch
+        id: create-branch
+        uses: ./.github/actions/neon-branch-create
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+          add_endpoint: false
+
+      - name: Attach the endpoint to the new branch
+        uses: ./.github/actions/neon-attach-endpoint-to-branch
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+          endpoint_id: ${{steps.endpoint-id.outputs.id}}
+          branch_id: ${{steps.create-branch.outputs.branch_id}}
+
+      - name: Wait for operations to be completed
+        uses: ./.github/actions/neon-wait-operations-to-finish
+        with:
+          api_key: ${{ secrets.NEON_STAGING_API_KEY }}
+          project_id: ${{matrix.project-id}}
+
```

Comment:
I do not fully understand why `create-branch` only is not enough here, and why do we need to re-attach endpoints?

Reply:
It's a good question, thanks, let me try to explain.
We have to add some settings to the endpoint (e.g. `TimeZone`, `compute_query_id` to `off`). We cannot add them by the API call due to restrictions (or not yet implemented). So updating the control plane database is the only way. But we cannot do this from our workflows, so we have to attach the pre-created endpoint to the new branch before testing and re-attach it to the main branch after.

Reply:
Discovered `default_endpoint_settings`, moved to using it